spec/openapi.yaml

```diff
@@ -3950,6 +3997,299 @@ components:
+                        suffix:
+                          description: Street number suffix
+                          type: string
+                          nullable: true
+                    zipCode:
+                      description: Zip code
+                      type: string
+                    city:
+                      description: City
+                      type: string
+                    countryCode:
+                      description: ISO 3166-1 alpha-2 country code
+                      type: string
+                      pattern: '^[A-Z]{2}$'
+                    phone:
+                      description: Phone number
+                      type: string
+                      nullable: true
+                    gender:
+                      description: Gender
+                      type: string
+                      nullable: true
+                      enum:
+                        - F
+                        - M
+                invoiceAddress:
+                  description: Invoice address of the customer
+                  type: object
+                  properties:
+                    id:
+                      description: Address identifier
+                      type: string
+                      format: uuid
+                    firstName:
+                      description: First name
+                      type: string
+                    lastName:
+                      description: Last name
+                      type: string
+                    company:
+                      description: Company name
+                      type: string
+                      nullable: true
+                    streetName:
+                      description: Street name
+                      type: string
+                    streetNumber:
+                      description: Street number
+                      type: string
+                    streetSuffix:
+                      description: Street number suffix
+                      type: string
+                      nullable: true
+                    zipCode:
+                      description: Zip code
+                      type: string
+                    city:
+                      description: City
+                      type: string
+                    countryCode:
+                      description: ISO 3166-1 alpha-2 country code
+                      type: string
+                      pattern: '^[A-Z]{2}$'
+                    phone:
+                      description: Phone number
+                      type: string
+                      nullable: true
+                    gender:
+                      description: Gender
+                      type: string
+                      nullable: true
+                      enum:
+                        - F
+                        - M
+        properties:
+          description: Transaction properties
+          type: array
+          items:
+            type: string
```

Comment:
Is there something else than a list of strings we could use here? Perhaps this can be better represented using a map-type. Example properties:
```
			"properties": {
				"minimum_age": "18",
				"shipping_region": "NL",
				"validation_callback_url": "https://olijf.myofflinestore.com/checkout/api/v1/orders/4dbac24c-0f4b-4dbf-8aac-beecbfc3ee30/validate"
			}
```

Reply:
I don't really know. All those values are optional. I can add it as example. But I don't know how we could represent this in the documentation itself.

Reply:
Perhaps we can model it as Free form objects (see https://swagger.io/docs/specification/data-models/dictionaries/). Alternatively we could consider to document each property individually.

Reply:
I prefer to have free-form then. the idea of the properties is that's not a fixed set of data, correct?

Reply:
Yeah fine by me, I agree that free-form matches best with the payment properties set :+1: